### DISCUSSION TITLE
How to utilize NetworkUtil::tcp_sum?

### DISCUSSION THREAD

**Discussion by Meraroo** [ANLAB-KAIST/KENSv3#62] [Project Q&A 2. Transport Layer (TCP)] [answered]
<img width="823" alt="스크린샷 2022-04-07 오후 5 00 15" src="https://user-images.githubusercontent.com/81629557/162150535-fd1d83f4-a701-4606-a60d-7da40a7ba28c.png">
<img width="375" alt="스크린샷 2022-04-07 오후 5 03 05" src="https://user-images.githubusercontent.com/81629557/162150737-ca2fb6c7-464f-4bd2-9d7a-8fa1a8b4146c.png">

I don't know how to use tcp_sum () function..
Can you give me a hint?

**Comment by overflow218** [comment]
I'm just a student. 
I'm not sure what you are curious about. 
If you mean how to use tcp_sum() function, 
You can refer src/Networking/E_networkUtil.cpp


uint16_t NetworkUtil::tcp_sum(uint32_t source, uint32_t dest,
                              const uint8_t *tcp_seg, size_t length) {
  if (length < 20)
    return 0;
  struct pseudoheader pheader;
  pheader.source = source;
  pheader.destination = dest;
  pheader.zero = 0;
  pheader.protocol = IPPROTO_TCP;
  pheader.length = htons(length);

  uint32_t sum = one_sum((uint8_t *)&pheader, sizeof(pheader));
  sum += one_sum(tcp_seg, length);
  sum = (sum & 0xFFFF) + (sum >> 16);
  return (uint16_t)sum;
}

**Comment by Meraroo** [comment, marked as answer]
Oh I found that initial value of checksum should be 0.. but I missed it.
